stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/usecase/SearchUsersByName.kt

```diff
@@ -0,0 +1,80 @@
+package io.getstream.chat.android.livedata.usecase
+
+import androidx.annotation.VisibleForTesting
+import io.getstream.chat.android.client.api.models.QuerySort
+import io.getstream.chat.android.client.api.models.QueryUsersRequest
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.call.CoroutineCall
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+
+public class SearchUsersByName internal constructor(private val chatDomainImpl: ChatDomainImpl) {
+
+    @VisibleForTesting
+    internal val defaultUsersQueryFilter by lazy {
+        Filters.and(
+            Filters.ne("name", ""),
+            Filters.ne("id", chatDomainImpl.currentUser.id)
+        )
+    }
+
+    /**
+     * Perform api request with a search string as autocomplete if in online state. Otherwise performs search by name
+     * in local database.
+     *
+     * @param querySearch Search string used as autocomplete.
+     */
+    public operator fun invoke(querySearch: String, offset: Int, userLimit: Int): Call<List<User>> {
+        return CoroutineCall(chatDomainImpl.scope) {
+            if (chatDomainImpl.isOnline()) {
+                performOnlineSearch(querySearch, offset, userLimit)
+            } else {
+                performOfflineSearch(querySearch, offset, userLimit)
+            }
+        }
+    }
+
+    private suspend fun performOfflineSearch(querySearch: String, offset: Int, userLimit: Int): Result<List<User>> {
+        return if (querySearch.isEmpty()) {
+            Result(chatDomainImpl.repos.selectAllUsers(userLimit, offset))
+        } else {
+            Result(chatDomainImpl.repos.selectUsersLikeName(querySearch, userLimit, offset))
+        }
+    }
+
+    private suspend fun performOnlineSearch(querySearch: String, offset: Int, userLimit: Int): Result<List<User>> {
+        val filter = if (querySearch.isEmpty()) {
+            defaultUsersQueryFilter
+        } else {
+            Filters.and(
+                Filters.autocomplete(FIELD_NAME, querySearch),
+                Filters.ne(FIELD_ID, chatDomainImpl.currentUser.id)
+            )
+        }
+
+        return chatDomainImpl.client.queryUsers(
+            QueryUsersRequest(
+                filter = filter,
+                offset = offset,
+                limit = userLimit,
+                querySort = USERS_QUERY_SORT,
+                presence = true
```

Comment:
Should we allow to pass that in `invoke` method?

Reply:
There is no sense to pass QuerySort since searching by name, and by default ordering by name works (otherwise we can't support any other case in DB).
But for presence it makes sense

Reply:
agree, would be nice to mention about default filter in KDOC.
`presence` also won't work with `offline` results right?